src/register/RegistrationPage.jsx

```diff
@@ -1,158 +1,131 @@
 import {
-  COMPLETE_STATE, DEFAULT_STATE,
-  INVALID_NAME_REGEX, LETTER_REGEX, NUMBER_REGEX, PENDING_STATE, REGISTER_PAGE, VALID_EMAIL_REGEX,
+  COMPLETE_STATE, PENDING_STATE, REGISTER_PAGE,
 } from '../data/constants';
 import {
   getAllPossibleQueryParams, getTpaHint, getTpaProvider, isHostAvailableInQueryParams, setCookie,
 } from '../data/utils';
 
-const emailRegex = new RegExp(VALID_EMAIL_REGEX, 'i');
-const urlRegex = new RegExp(INVALID_NAME_REGEX);
-
+/**
+ * Main Registration Page component
+ */
 const RegistrationPage = (props) => {
+  const { formatMessage } = useIntl();
+  const dispatch = useDispatch();
+
+  const registrationEmbedded = isHostAvailableInQueryParams();
+  const platformName = getConfig().SITE_NAME;
+  const flags = {
+    showConfigurableEdxFields: getConfig().SHOW_CONFIGURABLE_EDX_FIELDS,
+    showConfigurableRegistrationFields: getConfig().ENABLE_DYNAMIC_REGISTRATION_FIELDS,
+    showMarketingEmailOptInCheckbox: getConfig().MARKETING_EMAILS_OPT_IN,
+  };
   const {
-    backedUpFormData,
-    backendCountryCode,
-    backendValidations,
-    fieldDescriptions,
     handleInstitutionLogin,
     institutionLogin,
-    optionalFields,
+  } = props;
+
+  const {
+    registrationFormData: backedUpFormData,
     registrationError,
-    registrationErrorCode,
+    registrationError: {
+      errorCode: registrationErrorCode,
+    } = {},
     registrationResult,
     shouldBackupState,
+    userPipelineDataLoaded,
     submitState,
+    validations,
+  } = useSelector(state => state.register);
+
+  const {
+    fieldDescriptions,
+    optionalFields,
     thirdPartyAuthApiStatus,
     thirdPartyAuthContext,
-    usernameSuggestions,
-    validationApiRateLimited,
-    // Actions
-    backupFormState,
-    setUserPipelineDetailsLoaded,
-    getRegistrationDataFromBackend,
-    userPipelineDataLoaded,
-    validateFromBackend,
-    clearBackendError,
-  } = props;
+    thirdPartyAuthContext: {
+      autoSubmitRegForm,
+      errorMessage: thirdPartyAuthErrorMessage,
+      finishAuthUrl,
+      currentProvider,
+      providers,
+      secondaryProviders,
+      pipelineUserDetails,
+    },
+  } = useSelector(state => state.commonComponents);
 
-  const { formatMessage } = useIntl();
-  const countryList = useMemo(() => getCountryList(getLocale()), []);
+  const backendValidations = useMemo(
+    () => getBackendValidations(registrationError, validations), [registrationError, validations],
+  );
   const queryParams = useMemo(() => getAllPossibleQueryParams(), []);
-  const registrationEmbedded = isHostAvailableInQueryParams();
-  const { cta, host } = queryParams;
   const tpaHint = useMemo(() => getTpaHint(), []);
-  const flags = {
-    showConfigurableEdxFields: getConfig().SHOW_CONFIGURABLE_EDX_FIELDS,
-    showConfigurableRegistrationFields: getConfig().ENABLE_DYNAMIC_REGISTRATION_FIELDS,
-    showMarketingEmailOptInCheckbox: getConfig().MARKETING_EMAILS_OPT_IN,
-  };
+  const { cta, host } = queryParams;
```

Comment:
nit: Should this also be moved above useMemo with the other constants?

Reply:
actually, queryParams are computed in useMemo, so these are not available above useMemos. Either we move all constants below useMemos or do it like this.
I didn't want to move all constants below useMemos. 
What is your suggestion?

Reply:
sure.